src/libxively/sft/xi_sft_revision.c

```diff
@@ -77,13 +77,13 @@ static xi_state_t xi_sft_revision_read_string_from_file( const char* const resou
 
     XI_CHECK_STATE( state );
 
-    xi_bsp_io_fs_close( resource_handle );
-
     /* allocate output buffer then copy revision buffer into output buffer */
     XI_ALLOC_BUFFER_AT( char, *string_read_out, buffer_size + 1, state );
 
     memcpy( *string_read_out, buffer, buffer_size );
 
+    xi_bsp_io_fs_close( resource_handle );
```

Comment:
my change sinks this more deeper: under the err_handling, but this can stay here, But during merge put this under the err_handling label pls, thx!